### ISSUE TITLE
[HOLD for payment 2023-09-27] [$1000] Link Request Money not work

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#26214]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Sign in on browser as User A 
2. Create a chat with multiple users
3. Click the "+" green button icon in the chat and select Split Bill
4. Copy the URL in the browser (should be https://staging.new.expensify.com/split/new) and send it to the user B
5. Login on browser as User B -> go to chat User A
6. Click on the link of Request Money user A just sent
7. Enter the amount and press Next
8. Notice the Amount page loads again

## Expected Result:
Navigate to confirm the requested amount

## Actual Result:
The screen to enter the amount of the Split Bill is displayed again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:**  v1.3.57-5

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/a0bd23d1-848a-4adc-9bb7-db93b459789f

https://github.com/Expensify/App/assets/115492554/1eae76c4-ee37-488f-8f78-64c76faae30c

https://github.com/Expensify/App/assets/115492554/0122c24f-96b9-41fc-8b5a-d5a9fc9a8852

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Le Thi Thu Thuy

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692093712051379

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01354f2b491ad407b0</li>
        <li>Upwork Job ID: 1696603285025492992</li>
        <li>Last Price Increase: 2023-09-05</li>
<li>Automatic offers: </li>
<ul>
<li>akamefi202 | Contributor | 26512355</li>
</ul></li>
    </ul>
</details>

**Comment by akamefi202** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Link Request Money does not work. It opens Amount page again if the user clicks "Next" button.

### What is the root cause of that problem?
If the user opens Amount page from FAB, it will call `resetMoneyRequestInfo` function.
https://github.com/Expensify/App/blob/cb98bec57e43cdfbeb356b342054a53e47e59758/src/libs/actions/IOU.js#L1760-L1763

But if the user opens Amount page from link, it won't call that function. The Amount page will be opened directly.
So Onyx data for Amount page won't be initialized.

https://github.com/Expensify/App/blob/cb98bec57e43cdfbeb356b342054a53e47e59758/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsPage.js#L87-L89
Then if the user clicks "Next" button, the app opens Participants page.
But it goes back to Amount page immediately because of missing initialization of data.
As a result, the user will see Amount page again.

### What changes do you think we should make in order to solve the problem?
https://github.com/Expensify/App/blob/cb98bec57e43cdfbeb356b342054a53e47e59758/src/pages/iou/steps/NewRequestAmountPage.js#L127-L131

I think we need to move above code to outside of `if (isEditing)` conditional.
So `resetMoneyRequestInfo` function will be called when the user opens Amount page from the link.
Currently, it's not being called because of `if (isEditing)` conditional.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by hayata-suenaga** [comment]
I haven't read the original bug description of this issue, but yes, we want to refactor the navigation logic of money-request-related screens eventually

Is the bug pressing one? in that case, it's worth implementing a temporary fix, but otherwise, we can wait for this refactoring

(although I don't know when we can get to that refactoring)

**Comment by Christinadobrzyn** [comment]
Thanks @hayata-suenaga! @ArekChr do you think we should wait on this one for refactoring based on https://github.com/Expensify/App/issues/26214#issuecomment-1703326483

**Comment by ArekChr** [comment]
Hi @hayata-suenaga, I don't think this bug is pressing one, It mostly works, but the user has to press the next two times. Also, I think this issue when the user opens the split bill link from the report is an edge case.

**Comment by ArekChr** [comment]
[Proposal](https://github.com/Expensify/App/issues/26214#issuecomment-1698207936) from @akamefi202 solves that problem effectively. Let's go with your solution

:ribbon: :eyes: :ribbon: C+ reviewed

**Comment by akamefi202** [comment]
Thank you for selecting my proposal. I will create PR in a short time.

**Comment by Christinadobrzyn** [comment]
reached out to reporter for an Upwork link - https://expensify.slack.com/archives/C049HHMV9SM/p1693944049927919?thread_ts=1692093712.051379&cid=C049HHMV9SM

**Comment by CuongXuanLe** [comment]
Reporter:
Your Expensify account email: thuy.lethithu@antbuddy.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by thienlnam** [comment]
@CuongXuanLe Please follow the exact format in https://github.com/Expensify/App/issues/26214#issuecomment-1707062348 for your details to be stored!

**Comment by CuongXuanLe** [comment]
Contributor details
Your Expensify account email: thuy.lethithu@antbuddy.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by thuyle04** [comment]
Contributor details
Your Expensify account email: thuy.lethithu@antbuddy.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by akamefi202** [comment]
@ArekChr 
https://github.com/Expensify/App/pull/26854
PR is ready. Could you please review?

**Comment by ArekChr** [comment]
The PR for this issue has been reverted, as Split Bill functionality has been moved from the FAB to the Request Money flow, and it introduced regression in the new flow.

**Comment by akamefi202** [comment]
@ArekChr I will create PR again in the new flow soon. I was still able to confirm the bug.

**Comment by Christinadobrzyn** [comment]
Not sure if payment is still on schedule. @thienlnam or @ArekChr do you have an update on this? or a review of this PR - https://github.com/Expensify/App/issues/26214#issuecomment-1729297263

If we are still on schedule what is the regression test steps for this?

**Comment by Christinadobrzyn** [comment]
It looks like there was a regression here but I'm not sure if it's been fixed. @ArekChr or @thienlnam can you provide an update on where we are with this?

**Comment by thienlnam** [comment]
Yeah looks like the original PR added a regression / didn't solve the bug so we're waiting on @ArekChr to review the new PR that the contributor put up

**Comment by ArekChr** [comment]
Hello, I was on sick leave last week, I will review this PR soon.

**Comment by akamefi202** [comment]
@ArekChr I'm not able to reproduce now. I think it is fixed now.

**Comment by Christinadobrzyn** [comment]
Asking QA to test again - https://expensify.slack.com/archives/C9YU7BX5M/p1696616853061489

**Comment by Christinadobrzyn** [comment]
Waiting to hear from QA - I can't reproduce this either.

**Comment by Christinadobrzyn** [comment]
QA [confirmed](https://expensify.slack.com/archives/C9YU7BX5M/p1697057548496769?thread_ts=1696616853.061489&cid=C9YU7BX5M) they can't reproduce this either so I think we're good here. 

So I think we're ready to pay out based on https://github.com/Expensify/App/issues/26214#issuecomment-1727191890?

If that's the case that would be... 

Payouts due:

Issue Reporter: $250 @thuyle04 (old payment structure)
Contributor: $250 (50% due to regression) @akamefi202 
Contributor+: NA

Eligible for 50% #urgency bonus? No due to regression

Upwork job [here](https://www.upwork.com/jobs/~01354f2b491ad407b0) for @akamefi202 
Upwork job [here](https://www.upwork.com/jobs/~01c8ed54f75760e5cc) for @thuyle04 

Sound right?

**Comment by akamefi202** [comment]
@Christinadobrzyn I think it is $500 because original price was $1000.

**Comment by Christinadobrzyn** [comment]
Ah, it looks like Upwork auto-paid $1,000 to @akamefi202 on Oct 3rd, are you able to reject that payment? I'm not exactly sure what to do in this case

![image](https://github.com/Expensify/App/assets/51066321/2a8db51b-81b6-4efc-8013-39161759fc14)

**Comment by thuyle04** [comment]
Contributor details
Your Expensify account email: [thuy.lethithu@antbuddy.com](mailto:thuy.lethithu@antbuddy.com)
Upwork Profile Link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by thuyle04** [comment]
@Christinadobrzyn  this is my upwork link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by akamefi202** [comment]
@Christinadobrzyn It's my fault. I didn't know that it will be auto-paid. How about receiving $500 less in next job?

**Comment by ArekChr** [comment]
Hi @ChristinaDobrzyn, It seems like the PR did not get merged. It was reverted, and the error is not reproducible anymore due to changes in the split bill feature. I'm thinking we go for a 25% payout of the agreed price, given the situation. What do you think?

**Comment by Christinadobrzyn** [comment]
Ah thank you for catching that @ArekChr, okay the payment would be as follows here - https://github.com/Expensify/App/issues/26214#issuecomment-1758542357

Can you double-check?

@akamefi202 let me see if I can track down some other projects you're working on and tell them to adjust the payment.

**Comment by akamefi202** [comment]
@Christinadobrzyn I agree. I will ask deduction when the issue is completed.

**Comment by Christinadobrzyn** [comment]
Sounds good! Nothing yet from @le about the job so going to keep this open.

**Comment by thuyle04** [comment]
@Christinadobrzyn There seems to be a mistake here, my github account is Le Thi Thu Thuy, not le. So it's possible to tag the wrong person, please check carefully and help me correct it. Thanks
![Screenshot from 2023-10-12 22-29-15](https://github.com/Expensify/App/assets/63705956/37215e8f-a377-480a-8fa5-c902fa4f09cd)

**Comment by thuyle04** [comment]
Contributor details
Your Expensify account email: thuy.lethithu@antbuddy.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01fcd9ed589276332f

**Comment by Christinadobrzyn** [comment]
Thanks @thuyle04 I thought that was a little strange - didn't think to look at the Slack post - thanks for showing that. 

Can you accept this job offer so I can pay you? https://www.upwork.com/jobs/~01c8ed54f75760e5cc

**Comment by Christinadobrzyn** [comment]
paid @thuyle04 based on https://github.com/Expensify/App/issues/26214#issuecomment-1758542357

So I think this is all set!

**Comment by akamefi202** [comment]
@Christinadobrzyn I will do refund once I paid. It's very soon.

**Comment by akamefi202** [comment]
@mallenexpensify Could you please pay here first? That $1000 payment was more than 3 weeks ago so that I don't have enough money for refund in Upwork now. I promise that I will do refund.

**Comment by Christinadobrzyn** [comment]
okay monitoring the Upwork job for details about the refund - https://www.upwork.com/nx/wm/workroom/34742476/details

**Comment by mallenexpensify** [comment]
I paid the other issue, please leave this open til the refund here is processed, thx

**Comment by Christinadobrzyn** [comment]
@akamefi202 can you let me know when you've issued the refund? Thanks!

**Comment by akamefi202** [comment]
@Christinadobrzyn I will do refund on Monday next week.

**Comment by mallenexpensify** [comment]
@akamefi202 can you please confirm the refund? it doesn't look like we've received  yet.  Thanks.
<img width="359" alt="image" src="https://github.com/Expensify/App/assets/22508304/fa6ea9e5-ab86-427f-88d7-b66d45f6f79b">

**Comment by akamefi202** [comment]
@Christinadobrzyn @mallenexpensify Sorry for late reply. I was OOO last week. I made the refund and I think we might close this issue now.

**Comment by mallenexpensify** [comment]
Thanks @akamefi202 , confirming job was refunded.
<img width="393" alt="image" src="https://github.com/Expensify/App/assets/22508304/433e60fd-9d63-4c38-846b-a1fadff74b6a">

Pretty sure we're good to close this.

**Comment by akamefi202** [comment]
@mallenexpensify @Christinadobrzyn Could you please finish the contract on Upwork too?

**Comment by Christinadobrzyn** [comment]
Thanks @mallenexpensify and @akamefi202 - I closed the contract in Upwork.
